src/main/java/org/spongepowered/common/data/processor/common/SpawnerUtils.java

```diff
@@ -0,0 +1,100 @@
+package org.spongepowered.common.data.processor.common;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityList;
+import net.minecraft.entity.passive.EntityPig;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.MobSpawnerBaseLogic;
+import net.minecraft.world.World;
+import org.spongepowered.api.entity.EntitySnapshot;
+import org.spongepowered.api.entity.EntityType;
+import org.spongepowered.api.entity.EntityTypes;
+import org.spongepowered.api.util.weighted.TableEntry;
+import org.spongepowered.api.util.weighted.WeightedObject;
+import org.spongepowered.api.util.weighted.WeightedSerializableObject;
+import org.spongepowered.api.util.weighted.WeightedTable;
+import org.spongepowered.common.entity.SpongeEntitySnapshot;
+import org.spongepowered.common.entity.SpongeEntitySnapshotBuilder;
+import org.spongepowered.common.registry.type.entity.EntityTypeRegistryModule;
+
+import java.util.Optional;
+
+public class SpawnerUtils {
+
+    public static Optional<Entity> createEntity(EntityType type, World world) {
+        Entity entity;
+        try {
+            entity = (Entity) type.getEntityClass().getConstructor(World.class).newInstance(world);
+        } catch (Exception e) {
+            // This shouldn't happen for well behaved classes...
+            return Optional.empty();
+        }
+
+        return Optional.of(entity);
+    }
+
+    @SuppressWarnings("unchecked")
+    public static Optional<WeightedSerializableObject<EntitySnapshot>> getNextSnapshot(MobSpawnerBaseLogic logic) {
+        int weight = 1;
+        if(logic.getRandomEntity() != null) {
+            weight = logic.getRandomEntity().itemWeight;
+        }
+
+        Class<? extends Entity> entityClass = EntityList.stringToClassMapping.getOrDefault(logic.getEntityNameToSpawn(), EntityPig.class);
```

Comment:
I assume getting the class from `EntityTypeRegistryModule` would use the existing solution?